mantidimaging/gui/widgets/spectrum_widgets/roi_selection_widget.py

```diff
@@ -28,31 +28,30 @@ def __init__(self, parent: QtWidgets.QWidget = None):
         layout.addWidget(self.roiDropdown)
         self.previous_selection = None
 
-    def _on_selection_changed(self):
+    def _on_selection_changed(self) -> None:
         """ Handle dropdown selection change and emit signal. """
         selected_roi = self.roiDropdown.currentText()
         if selected_roi != self.previous_selection:
             self.previous_selection = selected_roi
             self.selectionChanged.emit(selected_roi)
 
-    def update_roi_list(self, roi_names: list[str]):
-        """ Update the dropdown. """
+    def update_roi_list(self, roi_names: list[str]) -> None:
+        """ Update the dropdown and trigger selection change if needed. """
         current_selection = self.roiDropdown.currentText()
         self.roiDropdown.blockSignals(True)
         self.roiDropdown.clear()
-
         filtered_rois = [name for name in roi_names if str(name) != str(ROI_RITS)]
         self.roiDropdown.addItems(filtered_rois)
-
         if current_selection in filtered_rois:
             self.roiDropdown.setCurrentText(current_selection)
         elif filtered_rois:
             self.roiDropdown.setCurrentIndex(0)
 
         self.previous_selection = self.roiDropdown.currentText()
```

Comment:
`self.previous_selection = self.roiDropdown.currentText()` might need to be removed, it will be set in `_on_selection_changed()`